### DISCUSSION TITLE
node packages on client side?

### DISCUSSION THREAD

**Discussion by kran27** [rejetto/hfs#506] [Q&A]
I was trying to make a plugin for previewing office files, [this](https://www.npmjs.com/package/@cyntler/react-doc-viewer) looks like the best option, but I have absolutely no clue how to use it, or if i even can. i tried looking at sources for various plugins and the documentation, couldn't find anything helpful.

**Comment by rejetto** [comment]
that package is a bit complex, probably a collection of things, and i didn't understand much with a quick look.
One thing i saw it's that it takes a lot of space just to parse pdf, which is natively supported by browsers, so unless one has specific needs, it's a bit of a waste.
You may have understood already that node_modules is for node, not client side. Still, it can still be used for client side when you use tools to build your final client files. But in this case, you don't want "source" files to be part of your distribution, you will put only built files in the "dist" folder of the plugin. What I'm telling you is about programming for the browser, not specific to hfs.

**Comment by sajkofff** [comment]
Hello. I use this extension in my organization

[Office Editing for Docs, Sheets & Slides](https://chrome.google.com/webstore/detail/office-editing-for-docs-s/gbkeegbaiigmenfmjfclcdgdpimamgkj)

But in my case there are some difficulties. Due to the lack of Internet on the local network, you have to use portable browsers.
I join this issue. I would like HFS to support viewing office documents.
